app/code/Magento/InventoryLowStockNotification/Model/ResourceModel/Product/Lowstock/Collection.php

```diff
@@ -0,0 +1,199 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryLowStockNotification\Model\ResourceModel\Product\Lowstock;
+
+use Magento\Catalog\Api\Data\ProductInterface;
+use Magento\CatalogInventory\Api\StockConfigurationInterface;
+use Magento\Eav\Api\AttributeRepositoryInterface;
+use Magento\Framework\Data\Collection\Db\FetchStrategyInterface;
+use Magento\Framework\Data\Collection\EntityFactoryInterface;
+use Magento\Framework\DB\Adapter\AdapterInterface;
+use Magento\Framework\Event\ManagerInterface;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Model\ResourceModel\Db\AbstractDb;
+use Magento\Framework\Model\ResourceModel\Db\Collection\AbstractCollection;
+use Magento\Inventory\Model\ResourceModel\SourceItem\Collection as SourceItemCollection;
+use Magento\InventoryApi\Api\Data\SourceItemInterface;
+use Magento\InventoryLowStockNotification\Setup\Operation\CreateSourceConfigurationTable;
+use Magento\InventoryLowStockNotificationApi\Api\Data\SourceItemConfigurationInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ */
+class Collection extends SourceItemCollection
```

Comment:
Maybe we need to create custom collection instead of extending `SourceItemCollection`